src/main/java/com/github/cameltooling/lsp/internal/completion/modeline/CamelKModelineInsertionProcessor.java

```diff
@@ -0,0 +1,60 @@
+package com.github.cameltooling.lsp.internal.completion.modeline;
+
+import org.eclipse.lsp4j.CompletionItem;
+import org.eclipse.lsp4j.TextDocumentItem;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+public class CamelKModelineInsertionProcessor {
+
+    private final TextDocumentItem textDocumentItem;
+
+    public CamelKModelineInsertionProcessor(TextDocumentItem textDocumentItem) {
+        this.textDocumentItem = textDocumentItem;
+    }
+
+    public CompletableFuture<List<CompletionItem>> getInsertion() {
```

Comment:
I think it will better to reuse the terminology of the Language Server Protocol and call it getCompletion() instead of getInsertion()